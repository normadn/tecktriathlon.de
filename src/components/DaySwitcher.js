import React, { Component } from "react";
import { css } from "glamor";

class DaySwitcher extends Component {
  state = {
    selectedDay: 0
  };

  changeDay(day, index) {
    this.props.changeDay(day);
    this.setState({ selectedDay: index });
  }

  render() {
    // truncate slider width to 3 decimals
    const sliderWidth = Math.trunc(100000 / this.props.days.length) / 1000;

    const sliderCss = {
      position: "absolute",
      left: `${this.state.selectedDay * sliderWidth}%`,
      top: 0,
      height: "44px",
      backgroundColor: "#C81C2E",
      borderRadius: "44px",
      width: `${sliderWidth}%`,
      zIndex: 2,
      transition: "left 0.8s ease"
    };

    const navBarCss = css({
      display: "grid",
      gridTemplateColumns: `repeat(${this.props.days.length}, 1fr)`,
      width: "66%",
      "@media screen and (max-width: 760px)": { width: "100%" },
      maxWidth: "600px",
      margin: "20px auto 35px auto",
      backgroundColor: "#F6F6F6",
      height: "44px",
      position: "relative",
      zIndex: 2,
      border: "1px solid #e4e4e4",
      borderRadius: "44px",
      "> button.dayItem": {
        textAlign: "center",
        lineHeight: "44px",
        textTransform: "uppercase",
        fontWeight: "bold",
        cursor: "pointer",
        background: "none",
        border: "none",
        fontSize: "18px",
        zIndex: 3,
        transition: "color 1s ease"
      },
      "> button.dayItem.selectedDay": {
        color: "white"
      },
      "> button.dayItem:not(.selectedDay)": {
        color: "#4d4d4d"
      },
      "> button.dayItem:not(.selectedDay):hover": {
        color: "#C81C2E"
      }
    });

    const navItems = this.props.days.map((day, i) => (
      <button
        key={day}
        onClick={() => this.changeDay(day, i)}
        className={
          this.state.selectedDay === i ? "selectedDay dayItem" : "dayItem"
        }
      >
        Day {i + 1}
      </button>
    ));

    return (
      <div {...navBarCss}>
        {navItems}
        <div role="none presentation" css={{ ...sliderCss }} />
      </div>
    );
  }
}

export default DaySwitcher;
