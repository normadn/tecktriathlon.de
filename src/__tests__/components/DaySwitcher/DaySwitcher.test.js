import React from "react";
import DaySwitcher from "../../../components/DaySwitcher";
import { shallow } from "enzyme";

describe("DaySwitcher", () => {
  it("renders correctly", () => {
    expect(
      shallow(<DaySwitcher days={["2019-02-01", "2019-02-02", "2019-02-03"]} />)
    ).toMatchSnapshot();
  });
});
