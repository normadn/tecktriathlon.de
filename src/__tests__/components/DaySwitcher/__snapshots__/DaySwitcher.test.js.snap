// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DaySwitcher renders correctly 1`] = `
<div
  data-css-11lhp2j=""
>
  <button
    className="selectedDay dayItem"
    key="2019-02-01"
    onClick={[Function]}
  >
    Day 
    1
  </button>
  <button
    className="dayItem"
    key="2019-02-02"
    onClick={[Function]}
  >
    Day 
    2
  </button>
  <button
    className="dayItem"
    key="2019-02-03"
    onClick={[Function]}
  >
    Day 
    3
  </button>
  <div
    css={
      Object {
        "backgroundColor": "#C81C2E",
        "borderRadius": "44px",
        "height": "44px",
        "left": "0%",
        "position": "absolute",
        "top": 0,
        "transition": "left 0.8s ease",
        "width": "33.333%",
        "zIndex": 2,
      }
    }
    role="none presentation"
  />
</div>
`;
